### PR TITLE
swupdate: make hawkbit server ip configurable

### PR DESCRIPTION
**Pull request by fahadarslan**
Signed-off-by: Fahad Arslan <Fahad_Arslan@mentor.com>

### CODE REVIEW

**Comment by fahadarslan** [comment]
This is connected with
https://stash.alm.mentorg.com/projects/MEL/repos/mel-meta-mentor-swupdate/pull-requests/11/overview

@chombourger, @Noor-Ahsan kindly review!

File: meta-iot2000-bsp/recipes-support/swupdate/swupdate_%.bbappend [hunk 1]
```diff
@@ -22,6 +22,8 @@ do_install_append() {
     install -d ${D}${sysconfdir}
     install -m 644 ${WORKDIR}/hwrevision ${D}${sysconfdir}
     install -m 644 ${WORKDIR}/swupdate.cfg ${D}${sysconfdir}
+
+    sed -i "s/hawkbit/${HAWKBIT_SERVER_IP}/g" ${D}${systemd_unitdir}/system/swupdate.service
```
Comment (chombourger) [inline meta-iot2000-bsp/recipes-support/swupdate/swupdate_%.bbappend]: we may want to make the complete URL user-configurable

Comment (chombourger) [reply meta-iot2000-bsp/recipes-support/swupdate/swupdate_%.bbappend]: update the service file to have a clear pattern. e.g. @hawkbit_url@ and use sed to replace it with the HAWKBIT_SERVER_URL setting from local.conf (or its default value, which needs to be defined somewhere)

File: meta-iot2000-bsp/recipes-support/swupdate/swupdate_%.bbappend [hunk 2]
```diff
@@ -22,6 +23,9 @@ do_install_append() {
     install -d ${D}${sysconfdir}
     install -m 644 ${WORKDIR}/hwrevision ${D}${sysconfdir}
     install -m 644 ${WORKDIR}/swupdate.cfg ${D}${sysconfdir}
+
+    sed -e "s@\@hawkbit_url\@@${HAWKBIT_SERVER_URL}@g" ${WORKDIR}/swupdate.service.in > ${WORKDIR}/swupdate.service
```
Comment (chombourger) [inline meta-iot2000-bsp/recipes-support/swupdate/swupdate_%.bbappend:27]: would have use a different separator to avoid escaping (such as ,) but OK

**Comment by chombourger** [comment]
looks good - thanks

**Pull request merged**